### DISCUSSION TITLE
Metadata: Add support for theme-color media queries

### DISCUSSION THREAD

**Discussion by michaelcummings12** [vercel/next.js#45854] [Ideas]
### Describe the feature you'd like to request

Some browsers support the ability to change the `theme-color` based on a media query. For example, if the device prefers dark mode, the browser can display a different `theme-color` than if the device prefers light mode.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name/theme-color

### Describe the solution you'd like

The `themeColor` type on the `Metadata` type could be something like

```
interface ThemeColor {
     color: string;
     media?: string;
}

interface Metadata {
     ...,
     themeColor?: string | ThemeColor[] | null,
}
```

### Describe alternatives you've considered

It is possible to create custom metadata tags that handle this, but it would be really nice to have the new metadata features be comprehensive.

**Comment by yasinkavakli** [comment]
Hopefully this gets worked on soon, can't believe you guys shipped this on canary without this. Should be always the default to include support for ThemeColor imo.

**Reply by balazsorban44** [reply]
The browser support is relatively low right now, and less useful on Desktop devices as Chrome only utilizes it for installed PWAs. See https://caniuse.com/meta-theme-color

Hence, this has not been a priority.

**Comment by johanguse** [comment]
Lighthouse warn me about the meta theme-color

![image](https://github.com/vercel/next.js/assets/6184866/64034dd6-fcac-45aa-8bb4-a39a59597b69)

My metadata config

![image](https://github.com/vercel/next.js/assets/6184866/f80f9bfc-b695-4832-99a4-fac38444ef8e)

**Comment by poziomekk** [comment]
the actual code for now as nextjs 14 docs says:

```
export const metadata= {
    themeColor: "#000000",
};
```

**transform into:**

```
export const viewport = {
    themeColor: "#000000",
};
```

**Reply by MarcusDelvecchio** [reply]
cool could you please provide source?

**Reply by poziomekk** [reply]
sorry, found on other site/thread but it works ;)

**Comment by johanguse** [comment]
I found this https://nextjs.org/docs/app/api-reference/functions/generate-viewport
Before 14, it was part of metadata, but now it has its own API. Just like the Metadata
